### PR TITLE
make kernel server honor `--log-path` and listen to kernel logging events

### PR DESCRIPTION
**Pull request by brettfo**
The fix was two parts:

1. Call `Program.StartToolLogging()` in `CommandLineParser.cs`.
2. Enable events from the type `KernelBase` in `Program.cs`.

Everything else was to get a proper unit test.

### CODE REVIEW

File: Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs [hunk 1]
```diff
@@ -85,5 +86,21 @@ private static DirectoryInfo EnsureExists(this DirectoryInfo directory)
 
             return directory;
         }
+
+        public static async Task<bool> WaitForFiles(this DirectoryInfo directory, TimeSpan timeout, Func<FileInfo[], bool> trigger)
```
Comment (jonsequitur) [inline Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs]: It would be better to accept a `CancellationToken` and throw if it elapses.

Comment (jonsequitur) [reply Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs]: Given the way it's being used you might also change the return type to a `FileInfo` and return the specific file.

File: dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs [hunk 2]
```diff
@@ -146,5 +152,49 @@ public async Task jupyter_returns_error_if_connection_file_path_is_not_passed()
 
             testConsole.Error.ToString().Should().Contain("Required argument missing for command: jupyter");
         }
+
+        [Fact(Timeout = 45000)]
```
Comment (jonsequitur) [inline dotnet-interactive.Tests/CommandLine/CommandLineParserTests.cs]: The timeout is unnecessary.

File: dotnet-interactive/build-and-install-dotnet-interactive.ps1 [hunk 3]
```diff
@@ -3,11 +3,9 @@ $ErrorActionPreference = "Stop"
 
 $thisDir = Split-Path -Parent $PSCommandPath
 $toolLocation = ""
-$toolVersion = ""
 if (Test-Path 'env:DisableArcade') {
-    dotnet pack "$thisDir\dotnet-interactive.csproj" /p:Version=0.0.0
+    dotnet pack "$thisDir\dotnet-interactive.csproj"
```
Comment (jonsequitur) [inline dotnet-interactive/build-and-install-dotnet-interactive.ps1]: Why do away with the version number?

Installing the specific version ensures that we don't grab a version from a nuget feed if it's higher than the local build, or if the local build is missing.

File: Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs (hunk 1 above)
Comment (brettfo) [reply Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs]: Honestly, all of this is a hack to see if I can get the test to consistently pass in CI.  Once I can get it to pass I'll make it look nice.  Naturally it works perfectly on my local machine.

**Pull request merged**